packages/plugins/graphql/server/src/services/builders/response-collection.ts

```diff
@@ -15,24 +17,39 @@ export default ({ strapi }: Context) => {
      */
     buildResponseCollectionDefinition(contentType: Schema.ContentType) {
       const name = naming.getEntityResponseCollectionName(contentType);
-      const entityName = naming.getEntityName(contentType);
+      const typeName = naming.getTypeName(contentType);
+      const { resolvePagination } = getService('builders').get('content-api');
 
       return objectType({
         name,
-
         definition(t) {
-          t.nonNull.list.field('data', {
-            type: nonNull(entityName),
+          // NOTE: add edges & cursor based pagination to support the relay spec in a later version
 
+          t.nonNull.list.field('nodes', {
+            type: nonNull(typeName),
             resolve: pipe(prop('nodes'), defaultTo([])),
           });
 
-          t.nonNull.field('meta', {
-            type: RESPONSE_COLLECTION_META_TYPE_NAME,
-
-            // Pass down the args stored in the source object
-            resolve: prop('info'),
+          t.nonNull.field('pageInfo', {
+            type: PAGINATION_TYPE_NAME,
+            resolve: resolvePagination,
           });
+
+          if (strapi.plugin('graphql').config('v4CompatibilityMode', false)) {
+            t.nonNull.list.field('data', {
+              deprecation: 'Use `nodes` field instead',
+              type: nonNull(typeName),
+              resolve: pipe(prop('nodes'), defaultTo([])),
+            });
+
+            // TODO: only add if v4 compat is enabled
```

Comment:
Shouldn't this be higher? (also true for data)

Reply:
you mean the comment ? yes I forgot to delete it actually